### ISSUE TITLE
marker-layer attributes do not track object changes

### ISSUE THREAD

**Issue by tevch** [miguelcobain/ember-leaflet#119]
1. create ember object with store.createRecord
2. create marker-layer based on this object, with attribute alt=object.id
3. call object save()

result - alt attribute remains null.

object itself does receive an id from the server - if i put object.id in the marker popup - it properly shows autogenerated id

**Comment by miguelcobain** [comment]
@tevch If you look at the leaflet API, there is not method to update `alt`, like there is for other properties.
That is a creation-time only option.

So, this is basically an upstream issue.
You should either create an issue at leaflet or PR the feature.

**Comment by tevch** [comment]
Understood. Thanks a lot
what method would you recommend to store an object id associated with the marker?

**Comment by miguelcobain** [comment]
The question is a bit vague. There are many ways to do that.

**Comment by tevch** [comment]
Let me rephrase the question

i create a marker like this

{{marker-layer location=object.location title-object.title alt=object.id onClick="markerClickedHandler"}}

then i get events in the function

markerClickedHandler(event) {
    alert("Clicked marker with ID="+event.target.options.alt)
}

The only option i see to retrieve the clicked marker object ID is to pass it to some parameter in {{marker-layer}} 

I used alt, but as you explained, it is not a good good choice.

How else can i bind object to a marker?
I appreciate your response

**Comment by miguelcobain** [comment]
If you're looking for a way to get a reference to the clicked object in the action, ember's closure actions have an awesome way to do that:

```hbs
{{marker-layer 
  location=object.location 
  onClick=(action "markerClicked" object)}}
```

then, in the action you get the curried object as an argument:

```js
markerClicked(target, object) {
  // use object here
}
```

**Comment by tevch** [comment]
Closure actions!!

Thanks a lot
